sample/todo/common/src/commonMain/kotlin/moe/tlaster/common/scene/NoteEditScene.kt

```diff
@@ -28,8 +29,9 @@ fun NoteEditScene(
     onDone: () -> Unit = {},
     onBack: () -> Unit = {},
 ) {
-    val viewModel = viewModel(NoteEditViewModel::class, listOf(id)) {
-        NoteEditViewModel(id)
+    val savedSateHolder = LocalSavedStateHolder.current
+    val viewModel = viewModel(NoteEditViewModel::class, listOf(id, savedSateHolder)) {
+        NoteEditViewModel(id, savedSateHolder)
```

Comment:
I'm passing the `SavedStateHolder` manually here, I thought about making the factory function `viewModel` do it for us, but I'm not sure what's the best approach to take to avoid a breaking change with it, since we need to pass the component to the `ViewModel` constructor.
One way is to introduce a new base class like `SavedViewModel` and an additional function that creates it, WDYT?

Reply:
I think it's better to add a new class like `SavedViewModel` for easier integration.

Reply:
After thinking more about this, I think a new class `SavedViewModel` won't add a lot of value, since client apps would still need to pass the value to its constructor after inheriting from it.
I instead opted to just update the `viewModel` factory function to allow passing the `SavedStateHolder` in its `creator` function, this should be a non-breaking change for apps, and it would still make using it easier just like in the sample app, WDYT?

Reply:
Good point, update the `viewModel` is easier to use and does not require the client apps to change the base class to `SavedViewModel`, LGTM!